### ISSUE TITLE
Error: The terraform-provider-datadog_v3.11.0 plugin crashed, Plugin did not respond

### ISSUE THREAD

**Issue by seifrajhi** [DataDog/terraform-provider-datadog#2198]
### Datadog Terraform Provider Version

v3.11.0

### Terraform Version

v1.3.2

### What resources or data sources are affected?

all resources in my code  are affected which are many, for instance:

datadog_logs_archive, datadog_integration_aws, datadog_integration_aws_lambda_arn, datadog_logs_index and many others

### Terraform Configuration Files

```hcl
Example:


resource "datadog_logs_archive" "stage" {
  name         = "stage Archive"
  query        = "env: stage"
  include_tags = true

  s3_archive {
    bucket     = "xxxxxxx"
    path       = "/"
    account_id = "xxxxxxx"
    role_name  = "xxxxxx"
  }
}
```
```


### Relevant debug or panic output

 Error: Request cancelled
 
Error: Plugin did not respond
 
with datadog_logs_archive.xxxx
on archives.tf line 120, in resource "datadog_logs_archive" "xxxx":
20: resource "datadog_logs_archive" "xxxx" {

The plugin encountered an error, and failed to respond to the
plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more
details.


Stack trace from the terraform-provider-datadog_v3.11.0 plugin:
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1219a9b]
goroutine 1077 [running]:


google.golang.org/grpc.(*Server).processUnaryRPC(0xc000224fc0, 0x18278f8, 0xc000603b00, 0xc000bd3900, 0xc000511c20, 0x2173630, 0x0, 0x0, 0x0)
	google.golang.org/grpc@v1.32.0/server.go:1194 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc000224fc0, 0x18278f8, 0xc000603b00, 0xc000bd3900, 0x0)
	google.golang.org/grpc@v1.32.0/server.go:1517 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc0003b8260, 0xc000224fc0, 0x18278f8, 0xc000603b00, 0xc000bd3900)
	google.golang.org/grpc@v1.32.0/server.go:859 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.32.0/server.go:857 +0x1fd
Error: The terraform-provider-datadog_v3.11.0 plugin crashed!
This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

### Expected Behavior

_No response_

### Actual Behavior

_No response_

### Steps to Reproduce

_No response_

### Important Factoids

_No response_

### References

_No response_

**Comment by seifrajhi** [comment]
Here are more logs:

I took the logs from the CI pipeline, I cannot reproduce it now because I upgraded to newer version of `Datadog Terraform Provider Version` and my pull request is merged.

```
│ Error: Request cancelled
│ 
│   with module.team["operations/config.yaml"].datadog_logs_metric.index_logs_metric["monitoring"],
│   on ../../modules/team/monitors.tf line 1, in resource "datadog_logs_metric" "index_logs_metric":
│    1: resource "datadog_logs_metric" "index_logs_metric" {
│ 
│ The plugin.(*GRPCProvider).UpgradeResourceState request was cancelled.
╵
Stack trace from the terraform-provider-datadog_v3.11.0 plugin:
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1219a9b]
goroutine 1077 [running]:
github.com/DataDog/datadog-api-client-go/api/v2/datadog.(*LogsArchiveDestination).GetActualInstance(...)
	github.com/DataDog/datadog-api-client-go@v1.13.0/api/v2/datadog/model_logs_archive_destination.go:126
github.com/terraform-providers/terraform-provider-datadog/datadog.buildDestination(0x0, 0x1633f01, 0x5, 0x13acd60, 0xc000bdc830, 0x0, 0x0)
	github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_logs_archive.go:194 +0x5b
github.com/terraform-providers/terraform-provider-datadog/datadog.updateLogsArchiveState(0xc000bd9280, 0xc0008d97d8, 0x0, 0x0, 0x16)
	github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_logs_archive.go:120 +0x270
github.com/terraform-providers/terraform-provider-datadog/datadog.resourceDatadogLogsArchiveRead(0x181f748, 0xc0007ff740, 0xc000bd9280, 0x133ef20, 0xc00069c1c0, 0xc000c660e0, 0xc000741878, 0x40e198)
	github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_logs_archive.go:153 +0x31c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc000229420, 0x181f6d8, 0xc000bdf700, 0xc000bd9280, 0x133ef20, 0xc00069c1c0, 0x0, 0x0, 0x0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.0/helper/schema/resource.go:358 +0x17f
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc000229420, 0x181f6d8, 0xc000bdf700, 0xc000b83ad0, 0x133ef20, 0xc00069c1c0, 0xc000b337b8, 0x0, 0x0, 0x0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.0/helper/schema/resource.go:635 +0x1cb
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc0000bc018, 0x181f6d8, 0xc000bdf700, 0xc000bdf740, 0x1640f87, 0x12, 0x0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.0/helper/schema/grpc_provider.go:576 +0x47d
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc00060c180, 0x181f780, 0xc000bdf700, 0xc0007fe480, 0x0, 0x0, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:553 +0x322
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler(0x158e3e0, 0xc00060c180, 0x181f780, 0xc000a9dce0, 0xc0007fe420, 0x0, 0x181f780, 0xc000a9dce0, 0xc00092b2c0, 0x121)
	github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:344 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000224fc0, 0x18278f8, 0xc000603b00, 0xc000bd3900, 0xc000511c20, 0x2173630, 0x0, 0x0, 0x0)
	google.golang.org/grpc@v1.32.0/server.go:1194 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc000224fc0, 0x18278f8, 0xc000603b00, 0xc000bd3900, 0x0)
	google.golang.org/grpc@v1.32.0/server.go:1517 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc0003b8260, 0xc000224fc0, 0x18278f8, 0xc000603b00, 0xc000bd3900)
	google.golang.org/grpc@v1.32.0/server.go:859 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.32.0/server.go:857 +0x1fd
Error: The terraform-provider-datadog_v3.11.0 plugin crashed!
This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

**Comment by therve** [comment]
Hi,

Thanks for the logs. This error has been fixed in #1915 and should be in version 3.26.0. Please upgrade and let us know if the issue is still present. Thanks.
